### ISSUE TITLE
target distribution plot is restricted to train set support

### ISSUE THREAD

**Issue by jeromedockes** [probabl-ai/skore#3020]
in the hub for regression a kde plot of the target distribution in the test set is shown, but only over the range covered by the train set:

https://github.com/probabl-ai/skore/blob/4b5cee88285e18e450725c479218a7ecb90e23c9/skore/src/skore/_plugins/hub/report/cross_validation_report.py#L313

so values out of this range which are likely to be hard to predict and are informative about either a real target shift or an issue in the train/test split procedure are not shown

```python
import skore
import numpy as np
from sklearn.dummy import DummyRegressor
from sklearn.model_selection import KFold

rng = np.random.default_rng()
X = rng.normal(size=(100, 3))
y = rng.normal(size=100)
y.sort()

report = skore.evaluate(DummyRegressor(), X=X, y=y, splitter=KFold(3, shuffle=False))
skore.login()
```
